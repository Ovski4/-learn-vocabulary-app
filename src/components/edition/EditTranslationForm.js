import React from 'react';
import { connect } from 'react-redux';
import { translationUpdated } from '../../actions/translations';
import { StyleSheet, TextInput, Button, Text, View } from 'react-native';

const mapDispatchToProps = (dispatch) => ({
    handleTranslationUpdated: (translation) => dispatch(translationUpdated(translation))
});

const styles = StyleSheet.create({
    view: {
        borderTopWidth: 0.4,
        borderColor: '#d6d7da',
    },
    row: {
        flexDirection: 'row'
    },
    label: {
        width: '15%',
        margin: 5
    },
    input: {
        flex: 1,
        backgroundColor: '#eeeeee',
        borderRadius: 2,
        paddingLeft: 5,
        margin: 5
    },
    submitButtonWrapper: {
        marginLeft: '15%'
    },
    submitButton: {
        marginLeft: 15,
        marginRight: 5,
        marginTop: 5,
        marginBottom: 10
    }
});

class EditTranslationForm extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            word1: props.translation.word1,
            word2: props.translation.word2,
            createdAt: props.translation.createdAt
        };
    }

    handleSubmit = () => {
        this.props.handleTranslationUpdated({
            word1: this.state.word1,
            word2: this.state.word2,
            createdAt: this.state.createdAt
        });
        this.props.onUpdate();
    }

    handleChange = (word, text) => {
        var newWord = {};
        newWord[word] = text;
        this.setState((previousState) => {
            return Object.assign({}, previousState, newWord);
        });
    }

    render() {
        return (
            <View style={styles.view}>
                <View style={styles.row}>
                    <View style={styles.label}>
                        <Text>{this.props.labelWord1}</Text>
                    </View>
                    <View style={styles.input}>
                        <TextInput
                            underlineColorAndroid="transparent"
                            value={this.state.word1}
                            onChangeText={(text) => this.handleChange('word1', text)}
                        />
                    </View>
                </View>
                <View style={styles.row}>
                    <View style={styles.label}>
                        <Text>{this.props.labelWord2}</Text>
                    </View>
                    <View style={styles.input}>
                        <TextInput
                            underlineColorAndroid="transparent"
                            value={this.state.word2}
                            onChangeText={(text) => this.handleChange('word2', text)}
                        />
                    </View>
                </View>
                <View style={styles.submitButtonWrapper}>
                    <View style={styles.submitButton}>
                        <Button title="Submit" onPress={this.handleSubmit}/>
                    </View>
                </View>
            </View>
        );
    }
}

EditTranslationForm.defaultProps = {
    labelWord1: 'Word 1:',
    labelWord2: 'Word 2:',
    translation: {
        word1: '',
        word2: '',
        createdAt: null
    }
};

export default connect(
    null,
    mapDispatchToProps
)(EditTranslationForm);
