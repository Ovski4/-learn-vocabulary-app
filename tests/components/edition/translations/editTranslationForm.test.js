import reducer from '../../../../src/reducers/index';
import { createStore } from 'redux';
import React from 'react';
import EditTranslationForm from '../../../../src/components/edition/translations/editTranslationForm/connect';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux'

describe('EditTranslationForm component', () => {

    const getStore = (locale) => {
        return createStore(reducer, {
            config: {
                locale: locale
            }
        });
    };

    it('Should render the component with the english locale', () => {
        const rendered = renderer.create(
            <Provider store={getStore('en')}>
                <EditTranslationForm translation={{
                    id: '1',
                    createdAt: new Date('December 20 1995'),
                    word1: 'word1',
                    word2: 'word2',
                    tags: []
                }} />
            </Provider>
        ).toJSON();

        expect(rendered).toMatchSnapshot();
    });

});
