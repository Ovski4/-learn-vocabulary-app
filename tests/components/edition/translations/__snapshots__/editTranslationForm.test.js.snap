// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditTranslationForm component Should render the component with the english locale 1`] = `
<View
  style={
    Object {
      "borderColor": "#d6d7da",
      "borderTopWidth": 0.4,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "justifyContent": "center",
          "margin": 5,
          "width": "15%",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Word 1
        :
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#eeeeee",
          "borderRadius": 2,
          "flex": 1,
          "margin": 5,
          "paddingLeft": 5,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        defaultValue="word1"
        onChangeText={[Function]}
        style={
          Object {
            "fontSize": 16,
            "height": 32,
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "justifyContent": "center",
          "margin": 5,
          "width": "15%",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Word 2
        :
      </Text>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#eeeeee",
          "borderRadius": 2,
          "flex": 1,
          "margin": 5,
          "paddingLeft": 5,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        defaultValue="word2"
        onChangeText={[Function]}
        style={
          Object {
            "fontSize": 16,
            "height": 32,
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "justifyContent": "center",
          "margin": 5,
          "width": "15%",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Tags
        :
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "marginRight": 5,
        }
      }
    >
      <View>
        <View
          style={
            Object {
              "backgroundColor": "#eeeeee",
              "borderRadius": 2,
              "marginLeft": 5,
            }
          }
        >
          <View
            style={
              Object {
                "height": 32,
              }
            }
          >
            <RCTPicker
              items={
                Array [
                  Object {
                    "label": "Pick a tag",
                    "textColor": undefined,
                    "value": null,
                  },
                ]
              }
              onChange={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              selectedIndex={0}
              style={
                Array [
                  Object {
                    "height": 216,
                  },
                  undefined,
                ]
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "flexDirection": "row",
              "flexWrap": "wrap",
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginLeft": "15%",
      }
    }
  >
    <View
      style={
        Object {
          "marginBottom": 10,
          "marginLeft": 15,
          "marginRight": 5,
          "marginTop": 5,
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Submit
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
